overlord/snapstate/target.go

```diff
@@ -110,7 +116,25 @@ func (t *target) snapsup(st *state.State, opts Options) (SnapSetup, error) {
 		PlugsOnly:          len(t.info.Slots) == 0,
 		InstanceKey:        t.info.InstanceKey,
 		ExpectedProvenance: t.info.SnapProvenance,
-	}, nil
+	}, compsups, nil
+}
+
+// componentTarget represents the data needed to setup a component for installation.
+type componentTarget struct {
+	// setup is a partially initialized ComponentSetup struct that contains the
+	// data needed to find the component that will be installed.
+	setup ComponentSetup
+	// info contains the snap.ComponentInfo for the component to be installed.
+	info *snap.ComponentInfo
+}
+
+func (c *componentTarget) compsup() ComponentSetup {
```

Comment:
It is shared here since there will be another implementation of `UpdateGoal` which will also return a target.

Reply:
But I do not see any mention to `UpdateGoal` in this PR (or am I missing something?), maybe it will become clear in the next PR, but atm I find this a bit confusing, maybe this part of the change could go in next one?

Reply:
Sorry, I meant `InstallGoal`. Too many PRs in my head :slightly_smiling_face:. I can move this to the next PR, which includes an `InstallGoal` implementation that installs from local files.